python/pyspark/ml/regression.py

```diff
@@ -2391,7 +2404,7 @@ def __repr__(self):
 
 
 class _FactorizationMachinesParams(_PredictorParams, HasMaxIter, HasStepSize, HasTol,
-                                   HasSolver, HasSeed, HasFitIntercept, HasRegParam):
+                                   HasSolver, HasSeed, HasFitIntercept, HasRegParam, HasWeightCol):
```

Comment:
Why add `HasWeightCol`? Seems you don't add a default of weight col?

Reply:
@viirya Thanks for reviewing. I added ```HasWeightCol``` in ```FactorizationMachinesParams``` in Scala in https://github.com/apache/spark/pull/28960 when I added ```FMClassificationSummary```.  ```ClassificationSummary``` needs weightCol, https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/ClassificationSummary.scala#L48. ```FMClassifier/Regressor``` doesn't support instance weight yet, so I didn't add ```setWeightCol``` or set default value in ```FactorizationMachinesParams```.